### ISSUE TITLE
SocialiteProviders / Apple PHP 8.3 dependency issue

### ISSUE THREAD

**Issue by tbanov** [SocialiteProviders/Providers#1236]
https://github.com/SocialiteProviders/Apple/commit/6863d005f8777b959a826cf074eeb611913b805f

Looks like lcobucci/clock 3.0 is not working with PHP 8.3

https://packagist.org/packages/lcobucci/clock#3.0.0

**Comment by atymic** [comment]
Feel free to PR a fix.

**Comment by hamrak** [comment]
Yes, 3.0.0 is not, but latest 3.3.1 is (php: ~8.2.0 || ~8.3.0 || ~8.4.0) : 
[lcobucci/clock#3.3.1 ](https://packagist.org/packages/lcobucci/clock#3.3.1)
and in our Apple composer.json we have: 
`"lcobucci/clock": "^2.0 || ^3.0",`

 ^3.0 definition is ok, it will install 3.3.1 too
